iroha/src/crypto.rs

```diff
@@ -57,6 +58,22 @@ pub struct Signature {
     signature: Ed25519Signature,
 }
 
+impl Ord for Signature {
+    fn cmp(&self, other: &Self) -> Ordering {
+        if self.public_key == other.public_key {
+            self.signature.to_vec().cmp(&other.public_key.to_vec())
+        } else {
+            self.public_key.cmp(&other.public_key)
+        }
+    }
+}
+
+impl PartialOrd for Signature {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
```

Comment:
Let's keep traits impls under the impl

Reply:
Sure. Actually this impl is no longer needed so I removed it.